Add source for Noto Sans Nushü

arrowtype opened the pull request:
This adds the GlyphsApp source for https://github.com/LisaHuang2017/noto-sans-nushu by @LisaHuang2017.

Question:
- I see that quite a few of the Noto fonts with Glyphs sources are accompanied by separate files for `GDEF, GPOS, GSUB` features. Is there a typical procedure followed to produce these?

punchcutter commented:
The separate GDEF/GPOS/GSUB are Monotype format OpenType. There’s no need to have them in most cases, but for various reasons many of them were necessary at the time.

arrowtype commented:
Cool; thanks for that information, @punchcutter!

punchcutter commented:
This file needs a little cleanup and some added info. I did a quick check and added things like copyright, languagesystem, Non-plane 0 in unicodeRanges. I also removed all of the unnecessary layers.  
[NotoSansNushu.glyphs.zip](https://github.com/googlefonts/noto-source/files/4478968/NotoSansNushu.glyphs.zip)

arrowtype commented:
Thanks, that's very helpful!

I realize those are probably all fairly obvious steps, but are those requirements documented somewhere? It would be awesome if they were included in the FontBakery checks at some point. :)

arrowtype commented:
Actually, I'm seeing that `unicodeRanges` includes several ranges that aren't present in this specific source (e.g. Basic Latin). Are those referring to Noto in general, or should this only say `Non-Plane 0`?

punchcutter commented:
I don't think there are any specific requirements anywhere, but there have just been some general things done along the way over the years and years and years this project has been going. FontBakery is also a new thing because before we were using `noto_lint` from the `nototools` repo. `noto_lint` would have reported half the things I did, but I don't think there are checks anywhere for extraneous layers except in `fontmake` which will tell you when you run it. There were a ton of lines like 
```
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer Regular: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name
WARNING:glyphsLib.builder.builders.UFOBuilder:Noto Sans Nushu Regular: Glyph .notdef, layer SemiBold: Duplicate glyph layer name

```

One thing for sure is you should never export from Glyphs because it's meaningless since the actual fonts will be built with fontmake. Also all sources are being moved to UFO so a Glyphs source file isn't necessary anymore. 

Most of the fonts have more than just Non-Plane 0. I don't remember right now if there were specific problems, but I wouldn't be surprised if not having the other Unicode ranges might cause problems somewhere.

arrowtype commented:
Okay, thanks, that is helpful as well.

Yup, it makes sense to remove unused layers – those were just on whitespace characters, and not in actual Nushü glyphs.

I have built the TTF output with FontMake in this very simple build script: 

https://github.com/arrowtype/noto-sans-nushu/blob/21ee589227781531d0a60567b0f102193fe444d5/mastering/build.sh

But, I'm sure updating to your cleaned up source will improve the TTF further.

Wouldn't including drawn glyphs that are outside the Nushü range cause more problems than excluding them? For instance, if we did include Latin Basic glyphs here, it might be a problem if they were ever adjusted/improved in the main Noto Sans font. Or, is this separate from what information we include in the `unicodeRanges` parameter?

punchcutter commented:
`unicodeRanges` just tells software what kind of glyphs are in the font, but even if there are no such glyphs it doesn't really do anything. So there are no glyphs for Latin Basic, but there are still common glyphs like `space` which is not Non-Plane 0. So `unicodeRanges` just says "there's supposedly support for this script".  There are definitely cases in older software (and probably newer as well because people never learn) where this info is expected and I wouldn't be surprised if something somewhere failed if this was only set to Non-Plane 0.

twardoch commented:
I remember in early webfont days, CSS unicode-ranges support would not work in a browser (possibly IE) for a given range if a bit wasn't set.

arrowtype commented:
Thanks again for your helpful feedback, @punchcutter! I've updated the commit with the fixed source. Please let me know if there is anything else to do here (e.g. should I run noto_lint on this, or does someone else / someone internal do that?).

arrowtype commented:
Actually, running this through the FontBakery checks for Google Fonts in general, a few issues are reported that seemingly would be good to fix further. Of course, Noto has some slightly different standards, but I will sort this out before this should be merged.

**These two are obvious, and I'll fix them now:**

<details>
<summary>🔥 <b>FAIL:</b> Checking font version fields (head and name table).</summary>

* [com.google.fonts/check/font_version](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/head.html#com.google.fonts/check/font_version)
* 🔥 **FAIL** head version is '1.000' while name version string (for platform 3, encoding 1) is 'Version 2.000'. [code: mismatch]

</details>

<details>
<summary>⚠ <b>WARN:</b> Font contains .notdef as first glyph?</summary>

* [com.google.fonts/check/mandatory_glyphs](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/universal.html#com.google.fonts/check/mandatory_glyphs)
* ⚠ **WARN** Font should contain the .notdef glyph as the first glyph, it should not have a Unicode value assigned and should contain a drawing.

</details>
<br/>

**These I am uncertain on:**

Fontbakery version: 0.7.4

<details>
<summary>🔥 <b>FAIL:</b> Check copyright namerecords match license file.</summary>

* [com.google.fonts/check/name/license](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/googlefonts.html#com.google.fonts/check/name/license)
* 🔥 **FAIL** License file OFL.txt exists but NameID 13 (LICENSE DESCRIPTION) value on platform 3 (WINDOWS) is not specified for that. Value was: "This Font Software is licensed under the SIL Open Font License, Version 1.1. This Font Software is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the SIL Open Font License for the specific language, permissions and limitations governing your use of this Font Software." Must be changed to "This Font Software is licensed under the SIL Open Font License, Version 1.1. This license is available with a FAQ at: http://scripts.sil.org/OFL" [code: wrong]

</details>
<details>
<summary>🔥 <b>FAIL:</b> "License URL matches License text on name table?</summary>

* [com.google.fonts/check/name/license_url](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/googlefonts.html#com.google.fonts/check/name/license_url)
* 🔥 **FAIL** A known license URL must be provided in the NameID 14 (LICENSE INFO URL) entry. Currently accepted licenses are Apache or Open Font License. For a small set of legacy families the Ubuntu Font License may be acceptable as well. [code: no-license-found]

</details>

<details>
<summary>🔥 <b>FAIL:</b> Copyright notices match canonical pattern in fonts</summary>

* [com.google.fonts/check/font_copyright](https://font-bakery.readthedocs.io/en/latest/fontbakery/profiles/googlefonts.html#com.google.fonts/check/font_copyright)
* 🔥 **FAIL** Name Table entry: Copyright notices should match a pattern similar to: 'Copyright 2017 The Familyname Project Authors (git url)'
But instead we have got: 'Copyright 2020 Google Inc. All Rights Reserved.'

</details>

punchcutter commented:
That's odd. .notdef is the first glyph so not sure why that warning is there. Font version should also be fine. The fails also should be fine, but I didn't get any of those on my end. Most of that info I copied from a recent Noto font to be sure they match things like copyright, license, etc. `noto_lint` is barely being maintained and has a lot of hard-coded info in it which takes some work to update. I've been trying to keep on top of it, but it seems hardly anyone cares and fontbakery is the new thing. There are some good things about `noto_lint` that would be nice to get into fontbakery, but maybe it's not highest priority right now.

arrowtype commented:
> .notdef is the first glyph so not sure why that warning is there.

It was missing an outline, so I pasted that in from NotoSans-MM.glyphs

Version

This should be the font version for Noto Nushü (which would be `1.000`) rather than for Noto as a whole, correct? Or, are new Noto fonts set to `2.000`? I've set both a `1.000` in this latest push.

punchcutter commented:
Version 1.0 of all Noto is pretty terrible in general so if someone is using a 1.000 or 1.001 font they probably shouldn't be. Things got updated to 2.000 so even though this font is a new one I think it would make more sense for it to be in the same "new" range of 2.000, better ask @marekjez86 to be sure.

arrowtype commented:
Ahh, okay, that makes sense. Well, I'm happy to use whatever makes sense to Marek. I sent an email to him with that question; hopefully he is able to respond here.

marekjez86 commented:
Version 1.0 is fine -- this is a brand new script so it's not an issue.
However, when I build it though I get

INFO:fontmake.font_project:Building master UFOs and designspace from Glyphs
source

INFO:glyphsLib.classes:Parsing "src/NotoSansNushu2.glyphs" file into
<GSFont>

INFO:fontmake.font_project:Interpolating master UFOs from designspace

fontmake: Error: In 'src/NotoSansNushu2.glyphs' ->
'master_ufo/NotoSansNushu-Regular.designspace' ->
'master_ufo/NotoSansNushu-Regular.designspace': Preparing the Designspace
for interpolation failed: Can't generate UFOs from this Designspace because
there is no default master source at location 'Weight: 100.0'. Check that
all 'default' values of all axes together point to a single actual master
source. For axes with a mapping, the 'default' values should have an
'input="..."' map value, where the corresponding 'output="..."' value then
points to the master source.



I'll look at it in 20 min though


On Wed, Apr 15, 2020 at 12:23 PM arrowtype <notifications@github.com> wrote:

> Ahh, okay, that makes sense. Well, I'm happy to use whatever makes sense
> to Marek. I sent an email to him with that question; hopefully he is able
> to respond here.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlefonts/noto-source/pull/188#issuecomment-614233234>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADFUS75R23F7ACN7BAIJUWLRMYCUFANCNFSM4MIGICVQ>
> .
>

arrowtype commented:
Hmm, I'm not sure what that error might be coming from.

Are you building with 

`fontmake -o ttf -g <filepath>/src/NotoSansNushu/NotoSansNushu.glyphs`

or another build tool?

Here is the current script I'm using to build the output TTF for https://github.com/googlefonts/noto-fonts/pull/1710:

https://github.com/arrowtype/noto-sans-nushu/blob/8734e06e8eb6f56e2b6bba369e40843a7dcc715d/mastering/build.sh

Your error has two things that I wonder about (sorry if these seem like dumb questions):
- `src/NotoSansNushu2.glyphs` has a `2` that I don't expect. Is it the right file?
- There is only a single master and single instance (Regular, 400), in the source. Are you perhaps attempting to build a variable font from it?

marekjez86 commented:
I tried to generate variable (it should generate something :-)).

I also used  ./build from https://github.com/googlefonts/noto-source which
in turn calls shell function from util.sh which in turn generates the
following command

fontmake -g src/NotoSansNushu2.glyphs -o ttf otf -i

try (with the right source file name)

... in my case it generated the following error
...fontmake -g src/NotoSansNushu2.glyphs -o otf  -i

INFO:fontmake.font_project:Building master UFOs and designspace from Glyphs
source

INFO:glyphsLib.classes:Parsing "src/NotoSansNushu2.glyphs" file into
<GSFont>

INFO:fontmake.font_project:Interpolating master UFOs from designspace

Traceback (most recent call last):

  File "/Library/Frameworks/Python.framework/Versions/3.7/bin/fontmake",
line 8, in <module>

    sys.exit(main())

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/__main__.py",
line 425, in main

    project.run_from_glyphs(glyphs_path, **args)

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
line 672, in run_from_glyphs

    self.run_from_designspace(designspace_path, **kwargs)

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
line 889, in run_from_designspace

    **kwargs,

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
line 932, in _run_from_designspace_static

    expand_features_to_instances=expand_features_to_instances,

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
line 707, in interpolate_instance_ufos

    designspace, round_geometry=round_instances

  File
"/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/instantiator.py",
line 185, in from_designspace

    "Can't generate UFOs from this designspace: no default font."

fontmake.instantiator.InstantiatorError: Can't generate UFOs from this
designspace: no default font.

On Wed, Apr 15, 2020 at 1:14 PM arrowtype <notifications@github.com> wrote:

> Hmm, I'm not sure what that error might be coming from.
>
> Are you building with
>
> fontmake -o ttf -g <filepath>/src/NotoSansNushu/NotoSansNushu.glyphs
>
> or another build tool?
>
> Here is the current script I'm using to build the output TTF for
> googlefonts/noto-fonts#1710
> <https://github.com/googlefonts/noto-fonts/pull/1710>:
>
>
> https://github.com/arrowtype/noto-sans-nushu/blob/8734e06e8eb6f56e2b6bba369e40843a7dcc715d/mastering/build.sh
>
> Your error has two things that I wonder about (sorry if these seem like
> dumb questions):
>
>    - src/NotoSansNushu2.glyphs has a 2 that I don't expect. Is it the
>    right file?
>    - There is only a single master and single instance (Regular, 400), in
>    the source. Are you perhaps attempting to build a variable font from it?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlefonts/noto-source/pull/188#issuecomment-614256990>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADFUS76NK76KNK6UND2LIMDRMYIRHANCNFSM4MIGICVQ>
> .
>

marekjez86 commented:
2 in my file name is just to differentiate PR Nushu from the one that I
took from Lisa's github

On Wed, Apr 15, 2020 at 1:40 PM Marek Z Jeziorek 老马 <marekj@google.com>
wrote:

>
> I tried to generate variable (it should generate something :-)).
>
> I also used  ./build from https://github.com/googlefonts/noto-source which
> in turn calls shell function from util.sh which in turn generates the
> following command
>
> fontmake -g src/NotoSansNushu2.glyphs -o ttf otf -i
>
> try (with the right source file name)
>
> ... in my case it generated the following error
> ...fontmake -g src/NotoSansNushu2.glyphs -o otf  -i
>
> INFO:fontmake.font_project:Building master UFOs and designspace from
> Glyphs source
>
> INFO:glyphsLib.classes:Parsing "src/NotoSansNushu2.glyphs" file into
> <GSFont>
>
> INFO:fontmake.font_project:Interpolating master UFOs from designspace
>
> Traceback (most recent call last):
>
>   File "/Library/Frameworks/Python.framework/Versions/3.7/bin/fontmake",
> line 8, in <module>
>
>     sys.exit(main())
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/__main__.py",
> line 425, in main
>
>     project.run_from_glyphs(glyphs_path, **args)
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
> line 672, in run_from_glyphs
>
>     self.run_from_designspace(designspace_path, **kwargs)
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
> line 889, in run_from_designspace
>
>     **kwargs,
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
> line 932, in _run_from_designspace_static
>
>     expand_features_to_instances=expand_features_to_instances,
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/font_project.py",
> line 707, in interpolate_instance_ufos
>
>     designspace, round_geometry=round_instances
>
>   File
> "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/fontmake/instantiator.py",
> line 185, in from_designspace
>
>     "Can't generate UFOs from this designspace: no default font."
>
> fontmake.instantiator.InstantiatorError: Can't generate UFOs from this
> designspace: no default font.
>
> On Wed, Apr 15, 2020 at 1:14 PM arrowtype <notifications@github.com>
> wrote:
>
>> Hmm, I'm not sure what that error might be coming from.
>>
>> Are you building with
>>
>> fontmake -o ttf -g <filepath>/src/NotoSansNushu/NotoSansNushu.glyphs
>>
>> or another build tool?
>>
>> Here is the current script I'm using to build the output TTF for
>> googlefonts/noto-fonts#1710
>> <https://github.com/googlefonts/noto-fonts/pull/1710>:
>>
>>
>> https://github.com/arrowtype/noto-sans-nushu/blob/8734e06e8eb6f56e2b6bba369e40843a7dcc715d/mastering/build.sh
>>
>> Your error has two things that I wonder about (sorry if these seem like
>> dumb questions):
>>
>>    - src/NotoSansNushu2.glyphs has a 2 that I don't expect. Is it the
>>    right file?
>>    - There is only a single master and single instance (Regular, 400),
>>    in the source. Are you perhaps attempting to build a variable font from it?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/googlefonts/noto-source/pull/188#issuecomment-614256990>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ADFUS76NK76KNK6UND2LIMDRMYIRHANCNFSM4MIGICVQ>
>> .
>>
>

punchcutter commented:
The weight values are inconsistent between Master and Instance. They should be the same value. In most fonts with multiple weights the Regular is somewhere around 86-100. Here it's 400, but the Instance also needs to match. I'd set both to 90 or 100 with the instance weight class custom parameter set to 400.

marekjez86 commented:
More issues that I'd like to considered (or at least answered):

E <cmap/required> b'U+0000 is not mapped in cmap, but it should be (see
https://www.microsoft.com/typography/otspec/recom.htm).'

E <cmap/required> b'U+000D is not mapped in cmap, but it should be (see
https://www.microsoft.com/typography/otspec/recom.htm).'

E <cmap/script_required> b'The following 2 characters are missing from the
font: 0000, 000D.'

E <head/os2/fsselection> b'UseTypoMetrics bit in fsSelection is set, but
should be clear.'

E <paths/intersection> b"The glyph 'u1B1B9' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B1BD' has intersecting outlines:
start and end segments match: (293.0, 141.0)"

E <paths/intersection> b"The glyph 'u1B1BE' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B1DF' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B1E5' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B200' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B20C' has intersecting outlines:
start and end segments match: (453.0, 369.0)"

E <paths/intersection> b"The glyph 'u1B212' has intersecting outlines:
start and end segments match: (293.0, 141.0)"

E <paths/intersection> b"The glyph 'u1B21B' has intersecting outlines:
start and end segments match: (453.0, 369.0)"

E <paths/intersection> b"The glyph 'u1B21C' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B21F' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B22A' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B23A' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B23B' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B240' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B248' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B24B' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B251' has intersecting outlines:
start and end segments match: (471.0, 117.0)"

E <paths/intersection> b"The glyph 'u1B255' has intersecting outlines:
start and end segments match: (313.0, 52.0)"

E <paths/intersection> b"The glyph 'u1B268' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B26D' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B27B' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B282' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B287' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B289' has intersecting outlines:
start and end segments match: (217.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B2AB' has intersecting outlines:
start and end segments match: (336.0, 496.0)"

E <paths/intersection> b"The glyph 'u1B2B4' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B2BA' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B2BB' has intersecting outlines:
start and end segments match: (206.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B2CF' has intersecting outlines:
start and end segments match: (257.0, 45.0)"

E <paths/intersection> b"The glyph 'u1B2D7' has intersecting outlines:
start and end segments match: (614.0, 117.0)"

E <paths/intersection> b"The glyph 'u1B2F2' has intersecting outlines:
start and end segments match: (488.0, 41.0)"

E <paths/intersection> b"The glyph 'u1B2FA' has intersecting outlines:
start and end segments match: (690.0, 326.0)"

E <complex/gpos/missing> b'There is no GPOS table in the font.'

E <complex/gsub/missing> b'There is no GSUB table in the font.'

E <reachable> b'The following 2 glyphs are unreachable in the font: CR (2),
NULL (1).'

On Wed, Apr 15, 2020 at 2:05 PM Zachary Quinn Scheuren <
notifications@github.com> wrote:

> The weight values are inconsistent between Master and Instance. They
> should be the same value. In most fonts with multiple weights the Regular
> is somewhere around 86-100. Here it's 400, but the Instance also needs to
> match. I'd set both to 90 or 100 with the instance weight class custom
> parameter set to 400.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlefonts/noto-source/pull/188#issuecomment-614279425>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADFUS7Z7ZR7OTGWQY5PRE3TRMYOQNANCNFSM4MIGICVQ>
> .
>

punchcutter commented:
The file I sent fixed the missing encoding for CR and NULL. There is no GPOS or GSUB so that doesn't matter. Intersecting outlines isn't a problem. The Use Typo Metrics parameter can be deleted.

arrowtype commented:
Shoot, I had tried to copy all the changes from Zachary’s file into the source from Lisa’s repo, then use that. But, I should have used a proper diff to make sure I wasn’t missing anything important. 😐 Zachary, should I go back and diff now? I want to make sure we don’t miss any improvements, but I’m also hoping to understand everything being updated, in case it is useful later on. Sorry for the extra loop that caused.

I will update those missing code points and remove the `use typo metrics`, and diff to see whether there’s anything more that should be covered.

punchcutter commented:
I think that’s all.

marekjez86 commented:
Got it... thanks for the reply and explanations.
Marek

On Wed, Apr 15, 2020 at 3:47 PM Zachary Quinn Scheuren <
notifications@github.com> wrote:

> The file I sent fixed the missing encoding for CR and NULL. There is no
> GPOS or GSUB so that doesn't matter. Intersecting outlines isn't a problem.
> The Use Typo Metrics parameter can be deleted.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/googlefonts/noto-source/pull/188#issuecomment-614317452>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADFUS75QJFFGGU3KN4V5XATRMY2QDANCNFSM4MIGICVQ>
> .
>

arrowtype commented:
Okay, I just checked a diff of the two files in VS Code to be certain and found a couple of few diffs:
- my file had different widths for NULL (should have been `0`) and CR (should have been `260`)
- my file didn't have panose values for the master; just for the instance
- Zachary's file seems to have an outline issue in the .notdef (overlapping points), but it's possible I somehow introduced this on my end
- my file had the weight value mismatch between the master and instance. I don't think it makes a difference whether it's 400 or 100 so long as this matches between master & instance, but if other Noto sources use 100, it's reasonable to match that, so I did
- version numbers are different, but it sounds like `1.000` works fine

I removed`use typo metrics` in this file (it was there because Google Fonts standards suggest using it, but I don't think it should matter if hhea & win values match typo).

I've updated the latest commit and force pushed to keep this clean. Let me know if there's anything else I can do here. Thanks for all the insight on this one, @punchcutter & @marekjez86!

LisaHuang2017 commented:
It's so amazing to see people volunteerly help on this project! I learned some new stuff these days :D Thank you so much!
